examples/gadgets/withfilter/trace/network/network.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"os/signal"
+	"syscall"
+
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection/networktracer"
+	containerutils "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils/types"
+	tracernetwork "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/network/tracer"
+	tracernetworktypes "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/network/types"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/socketenricher"
+	tracercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/tracer-collection"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/utils/host"
+)
+
+const (
+	networkTracerName = "trace_network"
+)
+
+func main() {
+	if err := run(); err != nil {
+		fmt.Fprintf(os.Stderr, "error: %s", err.Error())
+		os.Exit(1)
+	}
+}
+
+func run() error {
+	logger := slog.New(slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{}))
+	slog.SetDefault(logger)
+
+	err := host.Init(host.Config{
+		AutoMountFilesystems: false,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to initialized host filesystem: %w", err)
+	}
+
+	containerSelector := containercollection.ContainerSelector{}
+	containerCollection := &containercollection.ContainerCollection{}
+
+	tracerCollection, err := tracercollection.NewTracerCollection(containerCollection)
+	if err != nil {
+		return fmt.Errorf("failed to construct tracer collection: %w", err)
+	}
+
+	err = tracerCollection.AddTracer(networkTracerName, containerSelector)
+	if err != nil {
+		return fmt.Errorf("failed to add network tracer to tracer collection: %w", err)
+	}
```

Comment:
As I mentioned in https://github.com/inspektor-gadget/inspektor-gadget/issues/2593#issuecomment-1988803525, this gadget performs the filtering in a different way, hence the tracercollection is not needed:

```diff
diff --git examples/gadgets/withfilter/trace/network/network.go examples/gadgets/withfilter/trace/network/network.go
index 78f89351a..167107746 100644
--- examples/gadgets/withfilter/trace/network/network.go
+++ examples/gadgets/withfilter/trace/network/network.go
@@ -27,15 +27,10 @@ import (
        tracernetwork "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/network/tracer"
        tracernetworktypes "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/network/types"
        "github.com/inspektor-gadget/inspektor-gadget/pkg/socketenricher"
-       tracercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/tracer-collection"
        "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
        "github.com/inspektor-gadget/inspektor-gadget/pkg/utils/host"
 )
 
-const (
-       networkTracerName = "trace_network"
-)
-
 func main() {
        if err := run(); err != nil {
                fmt.Fprintf(os.Stderr, "error: %s", err.Error())
@@ -57,18 +52,7 @@ func run() error {
        containerSelector := containercollection.ContainerSelector{}
        containerCollection := &containercollection.ContainerCollection{}
 
-       tracerCollection, err := tracercollection.NewTracerCollection(containerCollection)
-       if err != nil {
-               return fmt.Errorf("failed to construct tracer collection: %w", err)
-       }
-
-       err = tracerCollection.AddTracer(networkTracerName, containerSelector)
-       if err != nil {
-               return fmt.Errorf("failed to add network tracer to tracer collection: %w", err)
-       }
-
        opts := []containercollection.ContainerCollectionOption{
-               containercollection.WithTracerCollection(tracerCollection),
                containercollection.WithContainerFanotifyEbpf(),
                containercollection.WithLinuxNamespaceEnrichment(),
                containercollection.WithMultipleContainerRuntimesEnrichment(
```

The logic that enables container filtering is `networktracer.ConnectToContainerCollection(config)`.

Reply:
Great, I've removed these lines as well.